### ISSUE TITLE
flatten command remove the empty security properties

### ISSUE THREAD

**Issue by ifraixedes** [go-swagger/go-swagger#1396]
## Problem statement

If one of the files has the `security` property of a path, set to an empty array (i.e. `"security": []`), the property is removed, so it doesn't appear in the output file.

## Swagger specification
2

## Steps to reproduce

I used the file of the example https://github.com/go-swagger/go-swagger/blob/master/examples/generated/swagger.json and split one minor part with a `$ref`, specifically the `paths.pets.post.parameters[0]`, following there are the input files (2) and the output file.

name of the file `petstore.json`

```json
{
    "swagger": "2.0",
    "info": {
        "description": "This is a sample server Petstore server.\n\n[Learn about Swagger](http://swagger.wordnik.com) or join the IRC channel '#swagger' on irc.freenode.net.\n\nFor this sample, you can use the api key 'special-key' to test the authorization filters\n",
        "version": "1.0.0",
        "title": "Swagger Petstore",
        "termsOfService": "http://helloreverb.com/terms/",
        "contact": {
            "name": "apiteam@wordnik.com"
        },
        "license": {
            "name": "Apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0.html"
        }
    },
    "host": "petstore.swagger.wordnik.com",
    "basePath": "/v2",
    "schemes": [
        "http"
    ],
    "paths": {
        "/pets": {
            "post": {
              "tags": [
                "pet"
              ],
              "summary": "Add a new pet to the store",
              "description": "",
              "operationId": "addPet",
              "consumes": [
                "application/json",
                "application/xml"
              ],
              "produces": [
                "application/json",
                "application/xml"
              ],
              "parameters": [
                {  "$ref": "pet-object-body.json/" }
              ],
              "responses": {
                "405": {
                  "description": "Invalid input"
                }
              },
              "security": []
            },
            "put": {
              "tags": [
                "pet"
              ],
              "summary": "Update an existing pet",
              "description": "",
              "operationId": "updatePet",
              "consumes": [
                "application/json",
                "application/xml"
              ],
              "produces": [
                "application/json",
                "application/xml"
              ],
              "parameters": [
                {
                  "in": "body",
                  "name": "body",
                  "description": "Pet object that needs to be added to the store",
                  "required": false,
                  "schema": {
                    "$ref": "#/definitions/Pet"
                  }
                }
              ],
              "responses": {
                "400": {
                  "description": "Invalid ID supplied"
                },
                "404": {
                  "description": "Pet not found"
                },
                "405": {
                  "description": "Validation exception"
                }
              },
              "security": [
                {
                  "petstore_auth": [
                    "write_pets",
                    "read_pets"
                  ]
                }
              ]
            }
        },
        "/pets/findByStatus": {
          "get": {
            "tags": [
              "pet"
            ],
            "summary": "Finds Pets by status",
            "description": "Multiple status values can be provided with comma seperated strings",
            "operationId": "findPetsByStatus",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "query",
                "name": "status",
                "description": "Status values that need to be considered for filter",
                "required": false,
                "type": "array",
                "items": {
                  "type": "string"
                },
                "collectionFormat": "multi"
              }
            ],
            "responses": {
              "200": {
                "description": "successful operation",
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/definitions/Pet"
                  }
                }
              },
              "400": {
                "description": "Invalid status value"
              }
            },
            "security": [
              {
                "petstore_auth": [
                  "write_pets",
                  "read_pets"
                ]
              }
            ]
          }
        },
        "/pets/findByTags": {
          "get": {
            "tags": [
              "pet"
            ],
            "summary": "Finds Pets by tags",
            "description": "Muliple tags can be provided with comma seperated strings. Use tag1, tag2, tag3 for testing.",
            "operationId": "findPetsByTags",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "query",
                "name": "tags",
                "description": "Tags to filter by",
                "required": false,
                "type": "array",
                "items": {
                  "type": "string"
                },
                "collectionFormat": "multi"
              }
            ],
            "responses": {
              "200": {
                "description": "successful operation",
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/definitions/Pet"
                  }
                }
              },
              "400": {
                "description": "Invalid tag value"
              }
            },
            "security": [
              {
                "petstore_auth": [
                  "write_pets",
                  "read_pets"
                ]
              }
            ]
          }
        },
        "/pets/{petId}": {
          "get": {
            "tags": [
              "pet"
            ],
            "summary": "Find pet by ID",
            "description": "Returns a pet when ID < 10.  ID > 10 or nonintegers will simulate API error conditions",
            "operationId": "getPetById",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "path",
                "name": "petId",
                "description": "ID of pet that needs to be fetched",
                "required": true,
                "type": "integer",
                "format": "int64"
              }
            ],
            "responses": {
              "200": {
                "description": "successful operation",
                "schema": {
                  "$ref": "#/definitions/Pet"
                }
              },
              "400": {
                "description": "Invalid ID supplied"
              },
              "404": {
                "description": "Pet not found"
              }
            },
            "security": [
              {
                "api_key": []
              },
              {
                "petstore_auth": [
                  "write_pets",
                  "read_pets"
                ]
              }
            ]
          },
          "post": {
            "tags": [
              "pet"
            ],
            "summary": "Updates a pet in the store with form data",
            "description": "",
            "operationId": "updatePetWithForm",
            "consumes": [
              "application/x-www-form-urlencoded"
            ],
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "path",
                "name": "petId",
                "description": "ID of pet that needs to be updated",
                "required": true,
                "type": "string"
              },
              {
                "in": "formData",
                "name": "name",
                "description": "Updated name of the pet",
                "required": true,
                "type": "string"
              },
              {
                "in": "formData",
                "name": "status",
                "description": "Updated status of the pet",
                "required": true,
                "type": "string"
              }
            ],
            "responses": {
              "405": {
                "description": "Invalid input"
              }
            },
            "security": [
              {
                "petstore_auth": [
                  "write_pets",
                  "read_pets"
                ]
              }
            ]
          },
          "delete": {
            "tags": [
              "pet"
            ],
            "summary": "Deletes a pet",
            "description": "",
            "operationId": "deletePet",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "header",
                "name": "api_key",
                "description": "",
                "required": true,
                "type": "string"
              },
              {
                "in": "path",
                "name": "petId",
                "description": "Pet id to delete",
                "required": true,
                "type": "integer",
                "format": "int64"
              }
            ],
            "responses": {
              "400": {
                "description": "Invalid pet value"
              }
            },
            "security": [
              {
                "petstore_auth": [
                  "write_pets",
                  "read_pets"
                ]
              }
            ]
          }
        },
        "/stores/order": {
          "post": {
            "tags": [
              "store"
            ],
            "summary": "Place an order for a pet",
            "description": "",
            "operationId": "placeOrder",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "body",
                "name": "body",
                "description": "order placed for purchasing the pet",
                "required": false,
                "schema": {
                  "$ref": "#/definitions/Order"
                }
              }
            ],
            "responses": {
              "200": {
                "description": "successful operation",
                "schema": {
                  "$ref": "#/definitions/Order"
                }
              },
              "400": {
                "description": "Invalid Order"
              }
            }
          }
        },
        "/stores/order/{orderId}": {
          "get": {
            "tags": [
              "store"
            ],
            "summary": "Find purchase order by ID",
            "description": "For valid response try integer IDs with value <= 5 or > 10. Other values will generated exceptions",
            "operationId": "getOrderById",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "path",
                "name": "orderId",
                "description": "ID of pet that needs to be fetched",
                "required": true,
                "type": "string"
              }
            ],
            "responses": {
              "200": {
                "description": "successful operation",
                "schema": {
                  "$ref": "#/definitions/Order"
                }
              },
              "400": {
                "description": "Invalid ID supplied"
              },
              "404": {
                "description": "Order not found"
              }
            }
          },
          "delete": {
            "tags": [
              "store"
            ],
            "summary": "Delete purchase order by ID",
            "description": "For valid response try integer IDs with value < 1000. Anything above 1000 or nonintegers will generate API errors",
            "operationId": "deleteOrder",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "path",
                "name": "orderId",
                "description": "ID of the order that needs to be deleted",
                "required": true,
                "type": "string"
              }
            ],
            "responses": {
              "400": {
                "description": "Invalid ID supplied"
              },
              "404": {
                "description": "Order not found"
              }
            }
          }
        },
        "/users": {
          "post": {
            "tags": [
              "user"
            ],
            "summary": "Create user",
            "description": "This can only be done by the logged in user.",
            "operationId": "createUser",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "body",
                "name": "body",
                "description": "Created user object",
                "required": false,
                "schema": {
                  "$ref": "#/definitions/User"
                }
              }
            ],
            "responses": {
              "default": {
                "description": "successful operation"
              }
            }
          }
        },
        "/users/createWithArray": {
          "post": {
            "tags": [
              "user"
            ],
            "summary": "Creates list of users with given input array",
            "description": "",
            "operationId": "createUsersWithArrayInput",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "body",
                "name": "body",
                "description": "List of user object",
                "required": false,
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/definitions/User"
                  }
                }
              }
            ],
            "responses": {
              "default": {
                "description": "successful operation"
              }
            }
          }
        },
        "/users/createWithList": {
          "post": {
            "tags": [
              "user"
            ],
            "summary": "Creates list of users with given input array",
            "description": "",
            "operationId": "createUsersWithListInput",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "body",
                "name": "body",
                "description": "List of user object",
                "required": false,
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/definitions/User"
                  }
                }
              }
            ],
            "responses": {
              "default": {
                "description": "successful operation"
              }
            }
          }
        },
        "/users/login": {
          "get": {
            "tags": [
              "user"
            ],
            "summary": "Logs user into the system",
            "description": "",
            "operationId": "loginUser",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "query",
                "name": "username",
                "description": "The user name for login",
                "required": false,
                "type": "string"
              },
              {
                "in": "query",
                "name": "password",
                "description": "The password for login in clear text",
                "required": false,
                "type": "string"
              }
            ],
            "responses": {
              "200": {
                "description": "successful operation",
                "schema": {
                  "type": "string"
                }
              },
              "400": {
                "description": "Invalid username/password supplied"
              }
            }
          }
        },
        "/users/logout": {
          "get": {
            "tags": [
              "user"
            ],
            "summary": "Logs out current logged in user session",
            "description": "",
            "operationId": "logoutUser",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "responses": {
              "default": {
                "description": "successful operation"
              }
            }
          }
        },
        "/users/{username}": {
          "get": {
            "tags": [
              "user"
            ],
            "summary": "Get user by user name",
            "description": "",
            "operationId": "getUserByName",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "path",
                "name": "username",
                "description": "The name that needs to be fetched. Use user1 for testing.",
                "required": true,
                "type": "string"
              }
            ],
            "responses": {
              "200": {
                "description": "successful operation",
                "schema": {
                  "$ref": "#/definitions/User"
                }
              },
              "400": {
                "description": "Invalid username supplied"
              },
              "404": {
                "description": "User not found"
              }
            }
          },
          "put": {
            "tags": [
              "user"
            ],
            "summary": "Updated user",
            "description": "This can only be done by the logged in user.",
            "operationId": "updateUser",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "path",
                "name": "username",
                "description": "name that need to be deleted",
                "required": true,
                "type": "string"
              },
              {
                "in": "body",
                "name": "body",
                "description": "Updated user object",
                "required": false,
                "schema": {
                  "$ref": "#/definitions/User"
                }
              }
            ],
            "responses": {
              "400": {
                "description": "Invalid user supplied"
              },
              "404": {
                "description": "User not found"
              }
            }
          },
          "delete": {
            "tags": [
              "user"
            ],
            "summary": "Delete user",
            "description": "This can only be done by the logged in user.",
            "operationId": "deleteUser",
            "produces": [
              "application/json",
              "application/xml"
            ],
            "parameters": [
              {
                "in": "path",
                "name": "username",
                "description": "The name that needs to be deleted",
                "required": true,
                "type": "string"
              }
            ],
            "responses": {
              "400": {
                "description": "Invalid username supplied"
              },
              "404": {
                "description": "User not found"
              }
            }
          }
        }
    },
    "securityDefinitions": {
      "api_key": {
        "type": "apiKey",
        "name": "api_key",
        "in": "header"
      },
      "petstore_auth": {
        "type": "oauth2",
        "authorizationUrl": "http://petstore.swagger.wordnik.com/api/oauth/dialog",
        "flow": "implicit",
        "scopes": {
          "write_pets": "modify pets in your account",
          "read_pets": "read your pets"
        }
      }
    },
    "definitions": {
      "User": {
        "properties": {
          "id": {
            "type": "integer",
            "format": "int64"
          },
          "username": {
            "type": "string"
          },
          "firstName": {
            "type": "string"
          },
          "lastName": {
            "type": "string"
          },
          "email": {
            "type": "string"
          },
          "password": {
            "type": "string"
          },
          "phone": {
            "type": "string"
          },
          "userStatus": {
            "type": "integer",
            "format": "int32",
            "description": "User Status"
          }
        }
      },
      "Category": {
        "properties": {
          "id": {
            "type": "integer",
            "format": "int64"
          },
          "name": {
            "type": "string"
          }
        }
      },
      "Pet": {
        "required": [
          "name",
          "photoUrls"
        ],
        "properties": {
          "id": {
            "type": "integer",
            "format": "int64"
          },
          "category": {
            "$ref": "#/definitions/Category"
          },
          "name": {
            "type": "string",
            "example": "doggie"
          },
          "photoUrls": {
            "type": "array",
            "items": {
              "type": "string"
            }
          },
          "tags": {
            "type": "array",
            "items": {
              "$ref": "#/definitions/Tag"
            }
          },
          "status": {
            "type": "string",
            "description": "pet status in the store"
          }
        }
      },
      "Tag": {
        "properties": {
          "id": {
            "type": "integer",
            "format": "int64"
          },
          "name": {
            "type": "string"
          }
        }
      },
      "Order": {
        "properties": {
          "id": {
            "type": "integer",
            "format": "int64"
          },
          "petId": {
            "type": "integer",
            "format": "int64"
          },
          "quantity": {
            "type": "integer",
            "format": "int32"
          },
          "shipDate": {
            "type": "string",
            "format": "date-time"
          },
          "status": {
            "type": "string",
            "description": "Order Status"
          },
          "complete": {
            "type": "boolean"
          }
        }
      }
    }
}
```

name of the file `pet-object-body.json`

```json
{
  "in": "body",
  "name": "body",
  "description": "Pet object that needs to be added to the store",
  "required": false,
  "schema": {
    "$ref": "#/definitions/Pet"
  }
}
```

Command to execute: `goswagger flatten -o swagger.json petstore.json`

the output file `swagger.json`

```json
{
  "schemes": [
    "http"
  ],
  "swagger": "2.0",
  "info": {
    "description": "This is a sample server Petstore server.\n\n[Learn about Swagger](http://swagger.wordnik.com) or join the IRC channel '#swagger' on irc.freenode.net.\n\nFor this sample, you can use the api key 'special-key' to test the authorization filters\n",
    "title": "Swagger Petstore",
    "termsOfService": "http://helloreverb.com/terms/",
    "contact": {
      "name": "apiteam@wordnik.com"
    },
    "license": {
      "name": "Apache 2.0",
      "url": "http://www.apache.org/licenses/LICENSE-2.0.html"
    },
    "version": "1.0.0"
  },
  "host": "petstore.swagger.wordnik.com",
  "basePath": "/v2",
  "paths": {
    "/pets": {
      "put": {
        "consumes": [
          "application/json",
          "application/xml"
        ],
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "pet"
        ],
        "summary": "Update an existing pet",
        "operationId": "updatePet",
        "security": [
          {
            "petstore_auth": [
              "write_pets",
              "read_pets"
            ]
          }
        ],
        "parameters": [
          {
            "description": "Pet object that needs to be added to the store",
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/Pet"
            }
          }
        ],
        "responses": {
          "400": {
            "description": "Invalid ID supplied"
          },
          "404": {
            "description": "Pet not found"
          },
          "405": {
            "description": "Validation exception"
          }
        }
      },
      "post": {
        "consumes": [
          "application/json",
          "application/xml"
        ],
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "pet"
        ],
        "summary": "Add a new pet to the store",
        "operationId": "addPet",
        "parameters": [
          {
            "description": "Pet object that needs to be added to the store",
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/Pet"
            }
          }
        ],
        "responses": {
          "405": {
            "description": "Invalid input"
          }
        }
      }
    },
    "/pets/findByStatus": {
      "get": {
        "description": "Multiple status values can be provided with comma seperated strings",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "pet"
        ],
        "summary": "Finds Pets by status",
        "operationId": "findPetsByStatus",
        "security": [
          {
            "petstore_auth": [
              "write_pets",
              "read_pets"
            ]
          }
        ],
        "parameters": [
          {
            "type": "array",
            "items": {
              "type": "string"
            },
            "collectionFormat": "multi",
            "description": "Status values that need to be considered for filter",
            "name": "status",
            "in": "query"
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "$ref": "#/definitions/findPetsByStatusOKBody"
            }
          },
          "400": {
            "description": "Invalid status value"
          }
        }
      }
    },
    "/pets/findByTags": {
      "get": {
        "description": "Muliple tags can be provided with comma seperated strings. Use tag1, tag2, tag3 for testing.",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "pet"
        ],
        "summary": "Finds Pets by tags",
        "operationId": "findPetsByTags",
        "security": [
          {
            "petstore_auth": [
              "write_pets",
              "read_pets"
            ]
          }
        ],
        "parameters": [
          {
            "type": "array",
            "items": {
              "type": "string"
            },
            "collectionFormat": "multi",
            "description": "Tags to filter by",
            "name": "tags",
            "in": "query"
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "$ref": "#/definitions/findPetsByTagsOKBody"
            }
          },
          "400": {
            "description": "Invalid tag value"
          }
        }
      }
    },
    "/pets/{petId}": {
      "get": {
        "description": "Returns a pet when ID \u003c 10.  ID \u003e 10 or nonintegers will simulate API error conditions",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "pet"
        ],
        "summary": "Find pet by ID",
        "operationId": "getPetById",
        "security": [
          {
            "api_key": []
          },
          {
            "petstore_auth": [
              "write_pets",
              "read_pets"
            ]
          }
        ],
        "parameters": [
          {
            "type": "integer",
            "format": "int64",
            "description": "ID of pet that needs to be fetched",
            "name": "petId",
            "in": "path",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "$ref": "#/definitions/Pet"
            }
          },
          "400": {
            "description": "Invalid ID supplied"
          },
          "404": {
            "description": "Pet not found"
          }
        }
      },
      "post": {
        "consumes": [
          "application/x-www-form-urlencoded"
        ],
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "pet"
        ],
        "summary": "Updates a pet in the store with form data",
        "operationId": "updatePetWithForm",
        "security": [
          {
            "petstore_auth": [
              "write_pets",
              "read_pets"
            ]
          }
        ],
        "parameters": [
          {
            "type": "string",
            "description": "ID of pet that needs to be updated",
            "name": "petId",
            "in": "path",
            "required": true
          },
          {
            "type": "string",
            "description": "Updated name of the pet",
            "name": "name",
            "in": "formData",
            "required": true
          },
          {
            "type": "string",
            "description": "Updated status of the pet",
            "name": "status",
            "in": "formData",
            "required": true
          }
        ],
        "responses": {
          "405": {
            "description": "Invalid input"
          }
        }
      },
      "delete": {
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "pet"
        ],
        "summary": "Deletes a pet",
        "operationId": "deletePet",
        "security": [
          {
            "petstore_auth": [
              "write_pets",
              "read_pets"
            ]
          }
        ],
        "parameters": [
          {
            "type": "string",
            "name": "api_key",
            "in": "header",
            "required": true
          },
          {
            "type": "integer",
            "format": "int64",
            "description": "Pet id to delete",
            "name": "petId",
            "in": "path",
            "required": true
          }
        ],
        "responses": {
          "400": {
            "description": "Invalid pet value"
          }
        }
      }
    },
    "/stores/order": {
      "post": {
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "store"
        ],
        "summary": "Place an order for a pet",
        "operationId": "placeOrder",
        "parameters": [
          {
            "description": "order placed for purchasing the pet",
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/Order"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "$ref": "#/definitions/Order"
            }
          },
          "400": {
            "description": "Invalid Order"
          }
        }
      }
    },
    "/stores/order/{orderId}": {
      "get": {
        "description": "For valid response try integer IDs with value \u003c= 5 or \u003e 10. Other values will generated exceptions",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "store"
        ],
        "summary": "Find purchase order by ID",
        "operationId": "getOrderById",
        "parameters": [
          {
            "type": "string",
            "description": "ID of pet that needs to be fetched",
            "name": "orderId",
            "in": "path",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "$ref": "#/definitions/Order"
            }
          },
          "400": {
            "description": "Invalid ID supplied"
          },
          "404": {
            "description": "Order not found"
          }
        }
      },
      "delete": {
        "description": "For valid response try integer IDs with value \u003c 1000. Anything above 1000 or nonintegers will generate API errors",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "store"
        ],
        "summary": "Delete purchase order by ID",
        "operationId": "deleteOrder",
        "parameters": [
          {
            "type": "string",
            "description": "ID of the order that needs to be deleted",
            "name": "orderId",
            "in": "path",
            "required": true
          }
        ],
        "responses": {
          "400": {
            "description": "Invalid ID supplied"
          },
          "404": {
            "description": "Order not found"
          }
        }
      }
    },
    "/users": {
      "post": {
        "description": "This can only be done by the logged in user.",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Create user",
        "operationId": "createUser",
        "parameters": [
          {
            "description": "Created user object",
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/User"
            }
          }
        ],
        "responses": {
          "default": {
            "description": "successful operation"
          }
        }
      }
    },
    "/users/createWithArray": {
      "post": {
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Creates list of users with given input array",
        "operationId": "createUsersWithArrayInput",
        "parameters": [
          {
            "description": "List of user object",
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/createUsersWithArrayInputParamsBody"
            }
          }
        ],
        "responses": {
          "default": {
            "description": "successful operation"
          }
        }
      }
    },
    "/users/createWithList": {
      "post": {
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Creates list of users with given input array",
        "operationId": "createUsersWithListInput",
        "parameters": [
          {
            "description": "List of user object",
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/createUsersWithListInputParamsBody"
            }
          }
        ],
        "responses": {
          "default": {
            "description": "successful operation"
          }
        }
      }
    },
    "/users/login": {
      "get": {
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Logs user into the system",
        "operationId": "loginUser",
        "parameters": [
          {
            "type": "string",
            "description": "The user name for login",
            "name": "username",
            "in": "query"
          },
          {
            "type": "string",
            "description": "The password for login in clear text",
            "name": "password",
            "in": "query"
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "type": "string"
            }
          },
          "400": {
            "description": "Invalid username/password supplied"
          }
        }
      }
    },
    "/users/logout": {
      "get": {
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Logs out current logged in user session",
        "operationId": "logoutUser",
        "responses": {
          "default": {
            "description": "successful operation"
          }
        }
      }
    },
    "/users/{username}": {
      "get": {
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Get user by user name",
        "operationId": "getUserByName",
        "parameters": [
          {
            "type": "string",
            "description": "The name that needs to be fetched. Use user1 for testing.",
            "name": "username",
            "in": "path",
            "required": true
          }
        ],
        "responses": {
          "200": {
            "description": "successful operation",
            "schema": {
              "$ref": "#/definitions/User"
            }
          },
          "400": {
            "description": "Invalid username supplied"
          },
          "404": {
            "description": "User not found"
          }
        }
      },
      "put": {
        "description": "This can only be done by the logged in user.",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Updated user",
        "operationId": "updateUser",
        "parameters": [
          {
            "type": "string",
            "description": "name that need to be deleted",
            "name": "username",
            "in": "path",
            "required": true
          },
          {
            "description": "Updated user object",
            "name": "body",
            "in": "body",
            "schema": {
              "$ref": "#/definitions/User"
            }
          }
        ],
        "responses": {
          "400": {
            "description": "Invalid user supplied"
          },
          "404": {
            "description": "User not found"
          }
        }
      },
      "delete": {
        "description": "This can only be done by the logged in user.",
        "produces": [
          "application/json",
          "application/xml"
        ],
        "tags": [
          "user"
        ],
        "summary": "Delete user",
        "operationId": "deleteUser",
        "parameters": [
          {
            "type": "string",
            "description": "The name that needs to be deleted",
            "name": "username",
            "in": "path",
            "required": true
          }
        ],
        "responses": {
          "400": {
            "description": "Invalid username supplied"
          },
          "404": {
            "description": "User not found"
          }
        }
      }
    }
  },
  "definitions": {
    "Category": {
      "properties": {
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "name": {
          "type": "string"
        }
      }
    },
    "Order": {
      "properties": {
        "complete": {
          "type": "boolean"
        },
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "petId": {
          "type": "integer",
          "format": "int64"
        },
        "quantity": {
          "type": "integer",
          "format": "int32"
        },
        "shipDate": {
          "type": "string",
          "format": "date-time"
        },
        "status": {
          "description": "Order Status",
          "type": "string"
        }
      }
    },
    "Pet": {
      "required": [
        "name",
        "photoUrls"
      ],
      "properties": {
        "category": {
          "$ref": "#/definitions/Category"
        },
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "name": {
          "type": "string",
          "example": "doggie"
        },
        "photoUrls": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "status": {
          "description": "pet status in the store",
          "type": "string"
        },
        "tags": {
          "$ref": "#/definitions/petTags"
        }
      }
    },
    "Tag": {
      "properties": {
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "name": {
          "type": "string"
        }
      }
    },
    "User": {
      "properties": {
        "email": {
          "type": "string"
        },
        "firstName": {
          "type": "string"
        },
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "lastName": {
          "type": "string"
        },
        "password": {
          "type": "string"
        },
        "phone": {
          "type": "string"
        },
        "userStatus": {
          "description": "User Status",
          "type": "integer",
          "format": "int32"
        },
        "username": {
          "type": "string"
        }
      }
    },
    "createUsersWithArrayInputParamsBody": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/User"
      },
      "x-go-gen-location": "operations"
    },
    "createUsersWithListInputParamsBody": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/User"
      },
      "x-go-gen-location": "operations"
    },
    "findPetsByStatusOKBody": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/Pet"
      },
      "x-go-gen-location": "operations"
    },
    "findPetsByTagsOKBody": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/Pet"
      },
      "x-go-gen-location": "operations"
    },
    "petTags": {
      "type": "array",
      "items": {
        "$ref": "#/definitions/Tag"
      },
      "x-go-gen-location": "models"
    }
  },
  "securityDefinitions": {
    "api_key": {
      "type": "apiKey",
      "name": "api_key",
      "in": "header"
    },
    "petstore_auth": {
      "type": "oauth2",
      "flow": "implicit",
      "authorizationUrl": "http://petstore.swagger.wordnik.com/api/oauth/dialog",
      "scopes": {
        "read_pets": "read your pets",
        "write_pets": "modify pets in your account"
      }
    }
  }
}
```

## Environment
swagger version: 0.13.0
OS:   OSX and Linux X64

I used the published binaries

**Comment by fredbi** [comment]
Possible duplicate of #1281

**Comment by ifraixedes** [comment]
Thanks.

I will give it a try at some point, but meanwhile a new version isn't released, I may delay until then, because mostly we use (my team) the same version of go-swagger binary and not everybody has Go installed for compiling it, besides that we guarantee everybody using the same version.

**Comment by fredbi** [comment]
Tested with PR#1478, which fixes the problem (it was indeed a duplicate issue).

**Comment by fredbi** [comment]
Closed with #1478
